### PR TITLE
Issue #322: reviewed experimental features documentation

### PR DESCRIPTION
**Pull request by shannonxtreme**
# Description

Review of `experimental.rst` and `exp_label_sampling.md`. 

Partially fixes Issue #322 

## Type of change

- [x] This change is a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Comments

@ebonmati @agrimwood I added a couple of comments into `exp_label_sampling.md` and would love your feedback!

### CODE REVIEW

File: docs/source/docs/exp_label_sampling.md
Comment (YipengHu) [inline docs/source/docs/exp_label_sampling.md:24]: Space between consistent labels

**Review by YipengHu — COMMENTED** [review]
Looks good but the length per line looks funny (could be because I'm using an Android...) Can you double check with pre-commit?

**Comment by shannonxtreme** [comment]
> Looks good but the length per line looks funny (could be because I'm using an Android...) Can you double check with pre-commit?

Hey, I ran into issues using pre-commit. I just set everything up on Ubuntu 20.04 using Python 3.8 and when I ran `pre-commit run --all-files` I get the following: 

```
An unexpected error has occurred: CalledProcessError: command: ('/home/shannon/miniconda3/envs/deepreg/bin/python', '-mvirtualenv', '/home/shannon/.cache/pre-commit/repow2oackzi/py_env-python3.7', '-p', 'python3.7')
return code: 1
expected return code: 0
stdout:
    RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.7'
    
stderr: (none)
Check the log at /home/shannon/.cache/pre-commit/pre-commit.log
```
It looks like it's looking for python 3.7 when my version is 3.8. Should I just downgrade my python installation?

**Comment by YipengHu** [comment]
I'd use a virtual environment and install DeepReg which will give you correct versions of python and pre-commit. More details are in the doc. Let me know if you still struggle.

**Comment by mathpluscode** [comment]
> An unexpected error has occurred: CalledProcessError

Hi @shannonxtreme, for multiple reasons including
- easier to control python packages and their complicated dependencies
- no harm to the system itself
- easy to add/remove/modify

we recommend to use Miniconda https://docs.conda.io/en/latest/miniconda.html
to manage the py env, if you are not using it, could you give it a chance ;)

For the py version, yes we officially use 3.7, no tests have been done on 3.8 yet, 
so it'd be nice if you can add a conda env based on 3.7. Installation guides are here https://deepreg.readthedocs.io/en/latest/getting_started/install.html

Please let us know if you have any problems/questions.

And if you still have bugs on pre-commit, let us know ;) I googled your error, 
it might be related to the permissions? https://github.com/pre-commit/pre-commit/issues/1046

**Comment by shannonxtreme** [comment]
It looks like I doubled down, lol. I did install miniconda, then cloned the repo and installed the package, then installed precommit and tried to run precommit. 

I'll try and do what you said. Complete noob to all this! Thanks for your patience.

**Comment by shannonxtreme** [comment]
Weird, I installed python3.7 and it's still broken. I just uninstalled miniconda and I'm going to try a fresh install of DeepReg to see if that helps. 

How do I uninstall precommit? Is everything automatically uninstalled when conda is?

**Comment by mathpluscode** [comment]
> Weird, I installed python3.7 and it's still broken. I just uninstalled miniconda and I'm going to try a fresh install of DeepReg to see if that helps.
> 
> How do I uninstall precommit? Is everything automatically uninstalled when conda is?

Tbh, I'm not sure, `pre-commit uninstall` maybe ? I found it here https://github.com/pre-commit/pre-commit/issues/364, try google it?

Btw for your errors it would be helpful if you share the log / screenshot

**Comment by shannonxtreme** [comment]
I uninstalled python3 and it uninstalled most of ubuntus packages along with it lmao. Anyway, I uninstalled miniconda and python and reinstalled, and now it's all good! 

Are you guys submitting this to uni in September? Good luck!!!

**Comment by mathpluscode** [comment]
> I uninstalled python3 and it uninstalled most of ubuntus packages along with it lmao. Anyway, I uninstalled miniconda and python and reinstalled, and now it's all good!
> 
> Are you guys submitting this to uni in September? Good luck!!!

Hi, thx so much for your effort. Good to hear that it's working for you now.

We are having some active changes this week, would you mind merge master to your branch and solve the conflicts so that we can merge this branch once it's ready. Thx for your understanding ;)

File: docs/source/docs/exp_label_sampling.md
Comment (DeepRegNet) [inline docs/source/docs/exp_label_sampling.md:35]: <> is this not clear? any one of the available label dissimilarity measures.

Comment (YipengHu) [reply docs/source/docs/exp_label_sampling.md:35]: Sorry @shannonxtreme  - it was my question @YipengHu - forgot to switch account after admin.

Comment (shannonxtreme) [reply docs/source/docs/exp_label_sampling.md:35]: No worries! So to me in this context measures is a little too ambiguous. I first thought you meant an actual distance measure like a millimetre values. 

I'm assuming there are specific conditions for label similarity/dissimilarity and that's what measures in this case is supposed to mean. If so, I suggest changing it to "label dissimilarity conditions / criteria"

**Comment by YipengHu** [comment]
Yup, better to be more specific, the label dissimilarity can be set to zero or left completely in config file. Would you mind change it to something like: if it is specified with a positive value...?

**Comment by YipengHu** [comment]
> Yup, better to be more specific, the label dissimilarity can be set to zero or left completely in config file. Would you mind change it to something like: if it is specified with a positive value...? 

Well actually negative will also work, just opposite effect to disencourrage the label alignment...

**Comment by YipengHu** [comment]
Let's leave it as is. I'll approval it now we can come back to this.

**Comment by shannonxtreme** [comment]
> Let's leave it as is. I'll approval it now we can come back to this.

Yeah haha good luck with your submission!! Let me know when I'm ok to make changes once again :)

**Comment by YipengHu** [comment]
@shannonxtreme you still have conflicts with the master. can you have a quick look and merge them before merging this?

**Comment by shannonxtreme** [comment]
> 
> 
> @shannonxtreme you still have conflicts with the master. can you have a quick look and merge them before merging this?

@YipengHu For `experimental.rst`, I don't think the link to open a new issue is required in the `toctree` as we link to it in the text immediately preceding it, but I am not familiar enough with Sphinx to remove the link.

**Comment by YipengHu** [comment]
> > @shannonxtreme you still have conflicts with the master. can you have a quick look and merge them before merging this?
> 
> @YipengHu For `experimental.rst`, I don't think the link to open a new issue is required in the `toctree` as we link to it in the text immediately preceding it, but I am not familiar enough with Sphinx to remove the link.

I'm not 100% sure either. I will approve this, if the checks passes, please merge it. Thanks! @mathpluscode see if you want to remove this link after this PR.

**Review by YipengHu — APPROVED** [review]

**Comment by shannonxtreme** [comment]
Did it fail because of the comment markup in the file? I can remove and recommit

File: docs/source/docs/experimental.rst
Comment (mathpluscode) [inline docs/source/docs/experimental.rst:7]: this is rst file not md file, so the grammar is a bit different.

instead of 
```
open a `new issue <https://github.com/DeepRegNet/DeepReg/issues/new>`__.
```

we can write
```
`open a new issue`.
```

with 
```
.. _open a new issue: https://github.com/DeepRegNet/DeepReg/issues/new
```

or simply delete the line 11 raise an issue ;)

**Review by mathpluscode — CHANGES_REQUESTED** [review]
please correct the experimental rst file, thx

**Review by mathpluscode — APPROVED** [review]

**Comment by mathpluscode** [comment]
@shannonxtreme I'm afraid that we haven't merged your branch yet?

**Comment by shannonxtreme** [comment]
> @shannonxtreme I'm afraid that we haven't merged your branch yet?

damn. restored! sorry, lol. I saw green checkmarks and closed it

**Comment by mathpluscode** [comment]
> > @shannonxtreme I'm afraid that we haven't merged your branch yet?
> 
> damn. restored! sorry, lol. I saw green checkmarks and closed it

No worries. Oh I approved it and wanted to wait the CI to pass to make sure. Forgot to come back in time.

**Pull request merged**
